### PR TITLE
Dropdown button

### PR DESCRIPTION
**Pull request by Bjvanminnen**
We currently have the AssignToSection component on the course/unit overview pages.
image

For some upcoming work I need to have a button with similar behavior, but different child elements in the dropdown portion. This work is done in preparation for that, extracting a generic DropdownButton component that is used by AssignToSection (and will also be used by my future work).
image

### CODE REVIEW

File: apps/src/templates/DropdownButton.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    position: 'relative',
+    top: 3
+  },
+  dropdown: {
+    border: `1px solid ${color.charcoal}`,
+    position: 'absolute',
+    // without this, this will be below some content
+    zIndex: 1,
+  },
+  section: {
```
Comment (caleybrock) [inline apps/src/templates/DropdownButton.js]: is this only used for sections right now?

File: apps/src/templates/DropdownButton.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+              />
+            ))}
+          </div>
+        )}
+      </div>
+    );
+  }
+}
+
+export default Radium(DropdownButton);
```
Comment (caleybrock) [inline apps/src/templates/DropdownButton.js:140]: what do you need radium for in this component?

File: apps/src/templates/DropdownButton.js (hunk 1 above)
Comment (Bjvanminnen) [reply apps/src/templates/DropdownButton.js]: It had been (this was copied over), but the style name should probably be updated now. good catch.

File: apps/src/templates/DropdownButton.js (hunk 2 above)
Comment (Bjvanminnen) [reply apps/src/templates/DropdownButton.js:140]: `:hover`

File: apps/test/unit/templates/DropdownButtonTest.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  it('passes through onClick and closes dropdown', () => {
+    const wrapper = shallow(
+      <DropdownButton {...defaultProps}/>
+    );
+    wrapper.find('Button').simulate('click');
+    assert(wrapper.find('a').at(1).props().onClick);
+    wrapper.find('a').at(1).simulate('click');
+    assert(clickSpy.calledOnce);
+
+    // dropdown is clsoed
```
Comment (caleybrock) [inline apps/test/unit/templates/DropdownButtonTest.js]: closed typo

Comment (Bjvanminnen) [reply apps/test/unit/templates/DropdownButtonTest.js]: Can't even claim that's just the Canadian spelling.

**Pull request merged**